src/apps/Altinn.AccessManagement/src/Altinn.AccessManagement.Integration/Clients/ResourceRegistryClient.cs

```diff
@@ -128,9 +129,42 @@ public async Task<IDictionary<string, IEnumerable<BaseAttribute>>> GetSubjectRes
             return subjectResources;
         }
 
-        public Task<ConsentTemplate> GetConsentTemplate(string templateId, int? version, CancellationToken cancellationToken = default)
+        ///<inheritdoc/>
+        public async Task<ConsentTemplate> GetConsentTemplate(string templateId, int? version, CancellationToken cancellationToken = default)
         {
-            throw new NotImplementedException();
+            List<ConsentTemplate> templates = await GetConsentTemplates(string.Empty);
+            ConsentTemplate consentTemplate = templates.FirstOrDefault(t => t.Id.Equals(templateId, StringComparison.OrdinalIgnoreCase));
+            if (consentTemplate == null)
+            {
+                throw new Exception($"Consent template with id {templateId} and version {version} not found.");
+            }
+
+            return consentTemplate;
+        }
+
+        private async Task<List<ConsentTemplate>> GetConsentTemplates(string org)
+        {
+            // Temp location. Will be moved to CDN
+            string url = "https://raw.githubusercontent.com/Altinn/altinn-studio-docs/master/content/authorization/architecture/resourceregistry/consent_templates.json";
+
+            try
+            {
+                HttpResponseMessage response = await _httpClient.GetAsync(url);
+                response.EnsureSuccessStatusCode();
+                string consentTemplatesString = await response.Content.ReadAsStringAsync();
+                List<ConsentTemplate> consentTemplates = JsonSerializer.Deserialize<List<ConsentTemplate>>(consentTemplatesString, _serializerOptions);
+                // Filter out templates not permitted for this service owner
+                consentTemplates = [.. consentTemplates
+                    .Where(t =>
+                        t.RestrictedToServiceOwners == null
+                        || t.RestrictedToServiceOwners.Count == 0
+                        || t.RestrictedToServiceOwners.Contains(org, StringComparer.OrdinalIgnoreCase))];
+                return consentTemplates;
```

Comment:
Vi trenger vel ikke filtrere bort noe her, kun returnere deserialiserte templates 
```suggestion
                return JsonSerializer.Deserialize<List<ConsentTemplate>>(consentTemplatesString, _serializerOptions);
```